### PR TITLE
Update dependencies, golang and alpine. Replace packr with embed.

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Update dependencies and golang
- vendor github.com/gardener/gardener@v1.28.1
- vendor github.com/gardener/gardener-resource-manager@v0.25.1
- update golang to 1.16.6
- update alpine to 3.13.5
- replace packr with embed

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement developer
`github.com/gardener/gardener@v1.28.1` and `github.com/gardener/gardener-resource-manager@v0.25.1` have been vendored.
```

```improvement developer
golang has been updated to 1.16.6
```

```improvement operator
alpine base image has been updated to 3.13.5
```

```improvement developer 
`github.com/gobuffalo/packr` has been replaced with golang native `embed`.
```

```feature operator
It is now possible to specify the leader election resource lock via the command line flag `--leader-election-resource-lock` (defaults to `configmapsleases`) and the chart value `leaderElection.resourceLock`. Please be careful when changing the resource lock and always migrate via multilocks in order to prevent situations where multiple instances of the controller are running with leader election and thus acting on the same resources.
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -311,19 +311,14 @@ github.com/go-openapi/validate v0.19.5/go.mod h1:8DJv2CVJQ6kGNpFW6eV9N3JviE1C85n
...
 github.com/gobuffalo/flect v0.1.5/go.mod h1:W3K3X9ksuZfir8f/LrfVtWmCDQFfayuylOJ7sz/Fj80=
 github.com/gobuffalo/flect v0.2.0/go.mod h1:W3K3X9ksuZfir8f/LrfVtWmCDQFfayuylOJ7sz/Fj80=
 github.com/gobuffalo/flect v0.2.2 h1:PAVD7sp0KOdfswjAw9BpLCU9hXo7wFSzgpQ+zNeks/A=
 github.com/gobuffalo/flect v0.2.2/go.mod h1:vmkQwuZYhN5Pc4ljYQZzP+1sq+NEkK+lh20jmEmX3jc=
-github.com/gobuffalo/logger v1.0.0 h1:xw9Ko9EcC5iAFprrjJ6oZco9UpzS5MQ4jAwghsLHdy4=
 github.com/gobuffalo/logger v1.0.0/go.mod h1:2zbswyIUa45I+c+FLXuWl9zSWEiVuthsk8ze5s8JvPs=
-github.com/gobuffalo/packd v0.3.0 h1:eMwymTkA1uXsqxS0Tpoop3Lc0u3kTfiMBE6nKtQU4g4=
 github.com/gobuffalo/packd v0.3.0/go.mod h1:zC7QkmNkYVGKPw4tHpBQ+ml7W/3tIebgeo1b36chA3Q=
-github.com/gobuffalo/packr v1.30.1 h1:hu1fuVR3fXEZR7rXNW3h8rqSML8EVAf6KNm0NKO/wKg=
 github.com/gobuffalo/packr v1.30.1/go.mod h1:ljMyFO2EcrnzsHsN99cvbq055Y9OhRrIaviy289eRuk=
```
Comment (voelzmo) [inline go.sum:321]: any idea why these two `packr` entry still remain?

**Review by voelzmo — COMMENTED** [review]
More questions than requesting changes – I'm fine with this going in like it is. Thanks!

**Review by voelzmo — APPROVED** [review]

File: go.sum (hunk 1 above)
Comment (vpnachev) [reply go.sum:321]: No idea. It is not used by any 3rd party component and I was wondering the same. 
Asking go
```bash
go mod why github.com/gobuffalo/packr
# github.com/gobuffalo/packr
(main module does not need package github.com/gobuffalo/packr)
```

Does not provide a useful answer.

File: pkg/charts/charts_test.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ var _ = Describe("Chart package test", func() {
 				},
 				"config": map[string]interface{}{
 					"nodeSelector": map[string]string{
-						extensionsv1alpha1.CRINameWorkerLabel: extensionsv1alpha1.CRINameContainerD,
+						extensionsv1alpha1.CRINameWorkerLabel: "containerd",
```
[the remark this answers is not in the document]
Comment (vpnachev) [reply pkg/charts/charts_test.go]: Reverted it back to `extensionsv1alpha1.CRINameContainerD`.

**Comment by vpnachev** [comment]
Updated the RBACs to allow usage of `leases.coordination.k8s.io` resources for leader election in 16e8cf7688fb34d0359605d1c4cf9701811145a5.
The resource lock is also configurable via CLI flag and chart value, the default is `configmapsleases`. In one of the next versions it can be switched to `leases`.

**Comment by voelzmo** [comment]
/lgtm
Successfully used this in conjunction with https://github.com/gardener/gardener-extension-runtime-gvisor/pull/21/commits/90f2cc8fe8ccaf8777e90e5e41bbbcff8154c3f5 to start a gvisor pod in my dev-shoot.

**Review by danielfoehrKn — APPROVED** [review]
thanks. tested it successfully

**Pull request merged**